py/dml/codegen.py

```diff
@@ -890,6 +890,14 @@ def fix_printf(fmt, args, argsites, site):
                                               TPtr(TNamed('char',
                                                           const=True)))),
                               [mkAddressOf(site, arg)])
+        elif conversion == 'b':
+            argtype = safe_realtype(arg.ctype())
+            if not isinstance(argtype, TBool):
+                raise EFMTARGT(argsites[argi], arg,
+                                argi+1, "boolean")
+            conversion = 's'
```

Comment:
An alternative to doing this and relying on the line below (which can be ruined by bad flags/width/precision/length specification) is:
```
filtered_args.append(...)
filtered_fmt += "%s"
continue
```
Is that preferable?

Reply:
I think this is an extremely non-confusing way to do it, I'd say its fine

Reply:
Which one? :)